File: timus/1748.py
p = [2,3,5,7,11,13,17,19,23,31,37,41,43,47,53,61,67,71]
t = int(input())
for i in range(t):
    n = int(input())
    cur = 1
    pos = True
    cnt = [1 for i in range(len(p)+1)]
    ans = 1
    while cur*2 <= n:
        mx = 0
        k = 0
        ind = 0
        hehe = 0
        for i in range(len(p)):
            if (ans//cnt[i])*(cnt[i]+1) > k and cur*p[i] <= n:
                hehe += 1
                k = (ans//cnt[i])*(cnt[i]+1)
                mx = cur*p[i]
                ind = i
        if hehe == 0:
            break
        ans = k
        cur = mx
        cnt[ind] += 1
    print(cur,ans)

File: timus/1748.cpp
#include<bits/stdc++.h>
#define ll long long
using namespace std;

vector<ll> a = {2,3,5,7,11,13,17,19,23,29,31,37,41,43,47,53,59,61,67,71,73,79,83,89,97,101};
ll f[50];
ll x,cnt,ans,c,t;
void cal(ll cur, ll divisors = 1, int pos = 0, ll mx = 60){
    if(divisors > cnt)cnt = divisors,ans=cur;
    else if (divisors == cnt)ans = min(cur,ans);
    if(pos == a.size())return;
    for(int i = 1; i <= mx && cur <= x/a[pos]; ++i){
        cur*=a[pos];
        cal(cur,divisors*(i+1),pos+1,i);
    }
}

int main(){
    scanf("%lld",&t);
    while(t--){
        scanf("%lld",&x);
        cnt = 1,ans = 1,c=1;
        cal(1);
        printf("%lld %lld\n",ans,cnt);
    }
    return 0;
}